src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```diff
@@ -0,0 +1,321 @@
+            }
+        }
+
+        private static bool OperationUsesPreviewFeatures(IOperation operation, ConcurrentDictionary<ISymbol, bool> requiresPreviewFeaturesSymbols, [NotNullWhen(true)] out ISymbol? symbol)
+        {
+            symbol = GetOperationSymbol(operation);
+            if (symbol == null)
+            {
+                return false;
+            }
+
+            return OperationUsesPreviewFeatures(symbol, requiresPreviewFeaturesSymbols);
+        }
+
+        private static ISymbol? GetOperationSymbol(IOperation operation)
+            => operation switch
+            {
+                IInvocationOperation iOperation => iOperation.TargetMethod,
+                IObjectCreationOperation cOperation => cOperation.Constructor,
+                IPropertyReferenceOperation pOperation => pOperation.Property,
+                IFieldReferenceOperation fOperation => fOperation.Field,
+                IDelegateCreationOperation dOperation => dOperation.Type,
+                IEventReferenceOperation eOperation => eOperation.Member,
+                IUnaryOperation uOperation => uOperation.OperatorMethod,
+                IBinaryOperation bOperation => bOperation.OperatorMethod,
+                IArrayCreationOperation arrayCreationOperation => (arrayCreationOperation.Type as IArrayTypeSymbol)?.ElementType,
+                ICatchClauseOperation catchClauseOperation => catchClauseOperation.ExceptionType,
+                _ => null,
+            };
+
+        private static bool OperationUsesPreviewFeatures(ISymbol symbol, ConcurrentDictionary<ISymbol, bool> requiresPreviewFeaturesSymbols)
+        {
+            var containingSymbol = symbol.ContainingSymbol;
+            // Namespaces do not have attributes
+            while (containingSymbol is INamespaceSymbol)
+            {
+                containingSymbol = containingSymbol.ContainingSymbol;
+            }
+
+            if (containingSymbol != null)
+            {
+                if (HasPreviewAttribute(containingSymbol, requiresPreviewFeaturesSymbols))
+                {
+                    return true;
+                }
+            }
+
+            return HasPreviewAttribute(symbol, requiresPreviewFeaturesSymbols);
+        }
+
+        private static bool HasPreviewAttribute(ISymbol symbol, ConcurrentDictionary<ISymbol, bool> requiresPreviewFeaturesSymbols)
+        {
+            if (!requiresPreviewFeaturesSymbols.TryGetValue(symbol, out bool existing))
+            {
+                bool ret = false;
+                ImmutableArray<AttributeData> attributes = symbol.GetAttributes();
+                foreach (AttributeData attribute in attributes)
+                {
+                    string attributeName = attribute.AttributeClass.Name;
+                    if (attributeName == RequiresPreviewFeaturesAttribute)
+                    {
+                        ret = true;
+                        break;
+                    }
+                }
```

Comment:
Can you instead use this extension method? https://github.com/dotnet/roslyn-analyzers/blob/a0522afd52b25ad211298e308fd5bbc7734865e0/src/Utilities/Compiler/Extensions/ISymbolExtensions.cs#L595

Reply:
Added a TODO. Can't do this yet, because we need to consume the latest sdk to get the `RequiresPreviewFeature` attribute symbol.

Reply:
> Can't do this yet, because we need to consume the latest sdk to get the `RequiresPreviewFeature` attribute symbol.

@pgovind I believe this is outdated now?